Logging/ARKLogMessage.h

```diff
@@ -27,15 +27,23 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface ARKLogMessage : NSObject <NSCopying, NSSecureCoding>
 
+/// @warning Do not use this initializer.
+/// @raises NSInternalInconsistencyException This is not a valid initializer.
+- (instancetype)init;
```

Comment:
yeah, but before I didn't really know what I was doing, and now I do.

I'd say let's do the following: Create an `ARKMessage` protocol that `ARKLogMessage` implements. Mark `new` and `init` as unavailable, and make everything that currently takes an `ARKLogMessage` take an `id <ARKMessage>` instead. That way we don't need to rely on subclassing (which leads to epic grossness), but rather can rely instead on protocol implementation.